atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -154,14 +162,20 @@
     @Parameterized.Parameters(name = "{0}")
     public static Collection<Object[]> data() {
         Object[][] data = new Object[][] {
-                {SYNC, UnaryOperator.identity()},
-                {ASYNC, (UnaryOperator<Transaction>) GetAsyncDelegate::new}
+                {SYNC,
+                 (BiFunction<Transaction, AtomicInteger, Transaction>) (transaction, counter) -> transaction,
```

Comment:
Let's introduce an interface, the type parameters are getting in the way:

```java
interface WrapperWithCounter<T> {
  T apply(T toWrap, AtomicInteger counter);
}
```

Reply:
Wrapper.